--- SuperLU_DIST_6.0.0/INSTALL/Makefile~	2018-09-22 17:56:51.000000000 +0100
+++ SuperLU_DIST_6.0.0/INSTALL/Makefile	2018-11-23 11:57:07.731549805 +0000
@@ -15,9 +15,9 @@
 	@echo Testing machines parameters and timer
 	csh install.csh
 
-smach_dist.o: ../SRC/smach_dist.c ; $(CC) -c $<
-dmach_dist.o: ../SRC/dmach_dist.c ; $(CC) -c $<
-superlu_timer.o: ../SRC/superlu_timer.c ; $(CC) -c $<
+smach_dist.o: ../SRC/smach_dist.c ; $(CC) $(CFLAGS) -c $<
+dmach_dist.o: ../SRC/dmach_dist.c ; $(CC) $(CFLAGS) -c $<
+superlu_timer.o: ../SRC/superlu_timer.c ; $(CC) $(CFLAGS) -c $<
 
 .c.o:
 	$(CC) $(CFLAGS) -c $<
