From a841c9be1666056b2a6b720d27161aa244b562b1 Mon Sep 17 00:00:00 2001
From: Yang Liu <liuyangzhuan@gmail.com>
Date: Thu, 26 May 2022 13:52:57 -0700
Subject: [PATCH] fixing the multiple definition errors in fortran files

---
 FORTRAN/CMakeLists.txt | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/FORTRAN/CMakeLists.txt b/FORTRAN/CMakeLists.txt
index 1f14310..63ec920 100644
--- a/FORTRAN/CMakeLists.txt
+++ b/FORTRAN/CMakeLists.txt
@@ -2,7 +2,6 @@
 include_directories(${SuperLU_DIST_SOURCE_DIR}/SRC)
 include_directories(${SuperLU_DIST_BINARY_DIR}/FORTRAN)
 
-set(mod_sources superlu_mod.f90 superlupara.f90)
 set(sources
     superlu_c2f_wrap.c  # initialize precision-independent file
     superlupara.f90
@@ -16,10 +15,8 @@ if(enable_complex16)
   list(APPEND sources c2f_zcreate_matrix_x_b.c superlu_c2f_zwrap.c)
 endif()  
 
-# see https://gitlab.kitware.com/cmake/cmake/-/issues/17525, why we need to separate MOD files from other source files
-add_library(superlu_dist_fortran_mod OBJECT ${mod_sources})
-add_library(superlu_dist_fortran $<TARGET_OBJECTS:superlu_dist_fortran_mod> ${sources})
-add_library(superlu_dist_fortran-static STATIC $<TARGET_OBJECTS:superlu_dist_fortran_mod> ${sources})
+add_library(superlu_dist_fortran ${sources})
+add_library(superlu_dist_fortran-static STATIC ${sources})
 # set(targets superlu_dist_fortran)
 get_target_property(superlu_dist_version superlu_dist VERSION)
 get_target_property(superlu_dist_soversion superlu_dist SOVERSION)
